pkg/detectors/artifactory/artifactory.go

```diff
@@ -61,20 +64,24 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		}
 		resMatch := strings.TrimSpace(match[1])
 
-		s1 := detectors.Result{
-			DetectorType: detectorspb.DetectorType_ArtifactoryAccessToken,
-			Raw:          []byte(resMatch),
-			RawV2:        []byte(resMatch + resURLMatch),
-		}
+		client := s.getClient()
+
+		for _, URL := range s.Endpoints(resURLMatch) {
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_ArtifactoryAccessToken,
+				Raw:          []byte(resMatch),
+				RawV2:        []byte(resMatch + URL),
+			}
+
+			if verify {
+				isVerified, verificationErr := verifyArtifactory(ctx, client, URL, resMatch)
+				s1.Verified = isVerified
+				s1.SetVerificationError(verificationErr, resMatch)
+			}
 
-		if verify {
-			client := s.getClient()
-			isVerified, verificationErr := verifyArtifactory(ctx, client, resURLMatch, resMatch)
-			s1.Verified = isVerified
-			s1.SetVerificationError(verificationErr, resMatch)
+			results = append(results, s1)
```

Comment:
Shouldn't a verified token break the loop? A token can't be valid on more than one instance, AFAIK.

Reply:
The idea is to capture the result of all URLs we verify against even if the verification passes or not.

Reply:
@rgmz I believe these are access token. If yes, then Form jfrog artifcatory [documentation](https://jfrog.com/help/r/jfrog-platform-administration-documentation/access-tokens), 

>  
> Cross-instance authentication
> 
> Access tokens can be used for authentication, not only by the instance or cluster where they were created but also for other instances and clusters that are all part of the same circle of trust.

Reply:
Interesting. Just be careful: Artifactory cloud has aggressive rate limiting. 

More than a few failed auth in a minute and you won't be able to hit it anymore, in my experience.

Reply:
@rgmz If Artifactory has any rate limiting, I believe that should be on the same endpoint. Here the endpoint will be changing for each request unless user pass same endpoint multiple times.

Reply:
@kashifkhan0771 if an endpoint is found multiple times in a chunk, won't we query it once per occurrence? That could cause rate-limiting, right? It's unlikely but theoretically possible. (I don't know whether it's worth worrying about it.)

Reply:
@rosecodym good point! In that case I think it's good to add a check to verify that the endpoints should be unique only.

Reply:
@rosecodym As per our logic for match URLs:
```
resURLMatch := ""
	for _, URLmatch := range URLmatches {
		if len(URLmatch) != 2 {
			continue
		}

		resURLMatch = strings.TrimSpace(URLmatch[1])
	}
```
We only pick one URL from the matches. I think a duplicate endpoint can only occur if user provide same endpoint multiple times. 
Shall we add a unique check for that or we can ignore it as that should be handled from user end?